arangod/Aql/DistributeExecutor.cpp

```diff
@@ -77,6 +79,35 @@ auto DistributeExecutorInfos::getResponsibleClient(arangodb::velocypack::Slice v
   return shardId;
 }
 
+auto DistributeExecutorInfos::shouldDistributeToAll(arangodb::velocypack::Slice value) const
+    -> bool {
+  if (_satellites.empty()) {
+    // We can only distribute to all on Satellite Collections
+    return false;
+  }
+  auto id = value.get(StaticStrings::IdString);
+  if (!id.isString()) {
+    // We can only distribute to all if we can detect the collection name
+    return false;
+  }
+
+  // NOTE: Copy Paste code, shall be unified
```

Comment:
Now or later? I'm fine with later - means we'll do it before merging back to devel branch.